tuftool/src/add_role.rs

```diff
@@ -6,81 +6,81 @@ use crate::datetime::parse_datetime;
 use crate::error::{self, Result};
 use crate::source::parse_key_source;
 use chrono::{DateTime, Utc};
+use clap::Parser;
 use snafu::{OptionExt, ResultExt};
 use std::num::NonZeroU64;
 use std::path::PathBuf;
-use structopt::StructOpt;
 use tough::editor::{targets::TargetsEditor, RepositoryEditor};
 use tough::key_source::KeySource;
 use tough::schema::{PathHashPrefix, PathPattern, PathSet};
 use url::Url;
 
-#[derive(Debug, StructOpt)]
+#[derive(Debug, Parser)]
 pub(crate) struct AddRoleArgs {
     /// The role being delegated
-    #[structopt(short = "d", long = "delegated-role")]
+    #[clap(short = 'd', long = "delegated-role")]
     delegatee: String,
 
     /// Key files to sign with
-    #[structopt(short = "k", long = "key", required = true, parse(try_from_str = parse_key_source))]
+    #[clap(short = 'k', long = "key", required = true, parse(try_from_str = parse_key_source))]
     keys: Vec<Box<dyn KeySource>>,
 
     /// Expiration of new role file; can be in full RFC 3339 format, or something like 'in
     /// 7 days'
-    #[structopt(short = "e", long = "expires", parse(try_from_str = parse_datetime))]
+    #[clap(short = 'e', long = "expires", parse(try_from_str = parse_datetime))]
     expires: DateTime<Utc>,
 
     /// Version of targets.json file
-    #[structopt(short = "v", long = "version")]
+    #[clap(short = 'v', long = "version")]
     version: NonZeroU64,
 
     /// Path to root.json file for the repository
-    #[structopt(short = "r", long = "root")]
+    #[clap(short = 'r', long = "root")]
     root: PathBuf,
 
     /// TUF repository metadata base URL
-    #[structopt(short = "m", long = "metadata-url")]
+    #[clap(short = 'm', long = "metadata-url")]
     metadata_base_url: Url,
 
     /// Incoming metadata
-    #[structopt(short = "i", long = "incoming-metadata")]
+    #[clap(short = 'i', long = "incoming-metadata")]
     indir: Url,
 
     /// threshold of signatures to sign delegatee
-    #[structopt(short = "t", long = "threshold")]
+    #[clap(short = 't', long = "threshold")]
     threshold: NonZeroU64,
 
     /// The directory where the repository will be written
-    #[structopt(short = "o", long = "outdir")]
+    #[clap(short = 'o', long = "outdir")]
     outdir: PathBuf,
 
     /// The delegated paths
-    #[structopt(short = "p", long = "paths", conflicts_with = "path-hash-prefixes")]
+    #[clap(short = 'p', long = "paths", conflicts_with = "path-hash-prefixes")]
     paths: Option<Vec<PathPattern>>,
 
     /// The delegated paths hash prefixes
-    #[structopt(short = "hp", long = "path-hash-prefixes")]
+    #[clap(short = 'x', long = "path-hash-prefixes")]
```

Comment:
Seems like we should call this out somewhere. But I don't know where that would be. Since we're the main consumer, it probably isn't too important though.

Reply:
I called it out in the commit message, but we should also mention it in the next release notes. My understanding is delegated paths don't really get used though. I also suspect "hp" was never a valid short (being that it was two characters).